mlflow/metrics/genai/genai_metric.py

```diff
@@ -558,4 +604,104 @@ def eval_fn(
         version=version,
         metric_details=evaluation_context["eval_prompt"].__str__(),
         metric_metadata=metric_metadata,
+        genai_metric_args=genai_metric_args,
     )
+
+
+def _filter_by_field(df, field_name, value):
+    return df[df[field_name] == value]
+
+
+def _deserialize_genai_metric_args(args_dict):
+    mlflow_version_at_ser = args_dict.get("mlflow_version")
+    fn_name = args_dict.get("fn_name")
+    if fn_name is None or mlflow_version_at_ser is None:
+        raise MlflowException(
+            message="The artifact JSON file appears to be corrupted and cannot be deserialized. "
+            "Please regenerate the custom metrics and rerun the evaluation. "
+            "Ensure that the file is correctly formatted and not tampered with.",
+            error_code=PUBLIC_UNDOCUMENTED,
+        )
+
+    if mlflow_version_at_ser != VERSION:
+        warnings.warn(
+            f"The custom metric definitions were serialized using MLflow {mlflow_version_at_ser}. "
+            f"Deserializing them with the current version {VERSION} might cause mismatches. "
+            "Please ensure compatibility or consider regenerating the metrics "
+            "using the current version.",
+            UserWarning,
+            stacklevel=2,
+        )
+
+    args_dict.pop("mlflow_version")
+    args_dict.pop("fn_name")
+    if fn_name == make_genai_metric_from_prompt.__name__:
+        return make_genai_metric_from_prompt(**args_dict)
+
+    examples = args_dict["examples"]
```

Comment:
For future compatibility, what do you think about storing the callee function name that was used in order to not rely on the presence / absence of a property of the config? 
While it's not likely that we'll introduce a new arg to the API, it could happen, causing issues for loading of a stored metric if the present of this arg is part of the control logic flow.

Reply:
> storing the callee function name
as in? sorry didn't follow.

currently the callee function name is stored (yes as a property of the config), mostly because I didn't want to pass around the function name as a separate arg in all the upstream calls, hence package it inside the config together with mlflow version

Reply:
Sorry for not seeing that. If we have that defined and within the config, let's use that! :)